charmcraft/commands/build.py

```diff
@@ -412,10 +437,22 @@ def process(self, parsed_args):
         """Process the received options."""
         result = {}
         for opt in self._options:
-            meth = getattr(self, "validate_" + opt)
+            print(parsed_args)
+            meth = getattr(self, "validate_" + opt.replace("-", "_"))
             result[opt] = meth(getattr(parsed_args, opt, None))
         return result
 
+    def validate_bases_indices(self, bases_indices):
+        """Validate that bases index is valid."""
+        if not bases_indices:
+            return
+
+        for bases_index in bases_indices:
+            if bases_index < 0:
```

Comment:
At this point the config is already loaded. We can totally validate here that the value is not bigger than the confgured bases...

Reply:
Good point. I added config to Validator and combined the tests.